src/connectors/configure/connectConfigure.js

```diff
@@ -60,10 +76,14 @@ export default function connectConfigure(renderFn = noop, unmountFn = noop) {
       refine(helper) {
         return searchParameters => {
           // merge new `searchParameters` with the ones set from other widgets
-          const actualState = this.removeSearchParameters(helper.state);
-          const nextSearchParameters = enhanceConfiguration(actualState, {
-            getConfiguration: () => searchParameters,
-          });
+          const actualState = getInitialSearchParameters(
+            helper.state,
+            widgetParams
+          );
+          const nextSearchParameters = mergeSearchParameters(
+            actualState,
+            new algoliasearchHelper.SearchParameters(searchParameters)
```

Comment:
just like before, it won't be removed I think if you add a parameter which wasn't in widgetParams, can we add a test for that?

Reply:
Can you elaborate the test case?